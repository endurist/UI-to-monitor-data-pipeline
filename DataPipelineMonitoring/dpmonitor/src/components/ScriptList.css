.script-list {
    width: 350px;
}
