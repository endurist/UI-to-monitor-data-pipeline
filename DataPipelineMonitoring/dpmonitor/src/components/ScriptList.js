import React from 'react';
import ListGroup from 'react-bootstrap/ListGroup';
import './ScriptList.css';

function ScriptList({ onScriptSelect, scriptList, activeIndex }) {
    if (scriptList.length === 0) {
        return (
            <div className='script-list'>
                <p className='create-script-message'>
                    Create a script to begin
                </p>
            </div>
        );
    }
    return (
        <div>
            <ListGroup
                className='script-list'
                defaultActiveKey='#0'
                onClick={onScriptSelect}
                activeKey={`#${activeIndex}`}
            >
                {scriptList.map((item, i) => {
                    return (
                        <ListGroup.Item action href={`#${i}`} key={i}>
                            {item.name}
                        </ListGroup.Item>
                    );
                })}
            </ListGroup>
        </div>
    );
}

export default ScriptList;
