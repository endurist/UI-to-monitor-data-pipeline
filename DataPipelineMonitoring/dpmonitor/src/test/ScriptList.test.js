import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { shallow } from 'enzyme';

import ScriptList from '../components/ScriptList';
import ListGroup from 'react-bootstrap/ListGroup';

afterEach(cleanup);

const scripts = [
    { id: 0, name: 'script1', scriptText: 'hi' },
    { id: 1, name: 'script2', scriptText: 'hi2' },
];
const emptyScripts = [];

it('should take a snapshot', () => {
    const { asFragment } = render(
        <ScriptList scriptList={scripts} activeIndex={0} />
    );

    expect(asFragment(<ScriptList />)).toMatchSnapshot();
});

it('should render one ListGroup', () => {
    const wrapper = shallow(
        <ScriptList scriptList={scripts} activeIndex={0} />
    );
    expect(wrapper.find(ListGroup)).toHaveLength(1);
});

it('should render two ListGroup.Items', () => {
    const wrapper = shallow(
        <ScriptList scriptList={scripts} activeIndex={0} />
    );
    expect(wrapper.find(ListGroup.Item)).toHaveLength(2);
});

it('should render zero List Group, and zero ListGroup.Items with empty array', () => {
    const wrapper = shallow(
        <ScriptList scriptList={emptyScripts} activeIndex={0} />
    );
    expect(wrapper.find(ListGroup)).toHaveLength(0);
    expect(wrapper.find(ListGroup.Item)).toHaveLength(0);
});

it('should render one <p> with a create script message if scripts is empty', () => {
    const wrapper = shallow(
        <ScriptList scriptList={emptyScripts} activeIndex={0} />
    );
    expect(wrapper.find('.create-script-message')).toHaveLength(1);
});

it('renders the script names in the list', () => {
    const { getByText } = render(
        <ScriptList scriptList={scripts} activeIndex={0} />
    );
    expect(getByText('script1')).toBeInTheDocument();
    expect(getByText('script2')).toBeInTheDocument();
});
