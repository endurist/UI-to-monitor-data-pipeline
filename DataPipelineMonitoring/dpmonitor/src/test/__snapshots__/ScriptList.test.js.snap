// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="script-list list-group"
    >
      <a
        class="list-group-item list-group-item-action active"
        data-rb-event-key="#link0"
        href="#link0"
      >
        script1
      </a>
      <a
        class="list-group-item list-group-item-action"
        data-rb-event-key="#link1"
        href="#link1"
      >
        script2
      </a>
    </div>
  </div>
</DocumentFragment>
`;
